cli/command/telemetry.go

```diff
@@ -131,6 +132,7 @@ func (r *telemetryResource) defaultOptions() []resource.Option {
 			// OTEL processors may think the same process is restarting
 			// continuously.
 			semconv.ServiceInstanceID(uuid.Generate().String()),
+			semconv.ServiceVersion(version.Version),
```

Comment:
Unrelated change, but I realized `semconv.ServiceVersion` is probably the attribute we should be using to capture the binary version.

> It represents the version string of the service API or implementation.

Reply:
Is this code that will also be consumed/inherited by plugins? (in that case, do they need a different version to be used?). Note that the `version.Version` is set at compile time, so this would work for the CLI itself (binaries built with the `-X ...` options set), but won't be the case for library / module uses (in which case it would be `unknown-version`).

We also have `UserAgent()`, which combines some parts of version (including git-commit), but the same applies there if these variables are not set during compile time.

Reply:
That's a good point! I don't think this code in particular would be used by plugins, only by the main CLI process. This code is configuration for the main CLIs `telemetryResource` which we use to identify one CLI invocation, but plugins will generally configure their own (see Compose's [use](https://github.com/docker/compose/blob/ff20b641c7f1cce588faea476fbd1a6fdd953b36/internal/tracing/tracing.go#L107-L110)), but it
s good to check

Reply:
cc @milas @jsternberg

Reply:
With the current implementation, each plugin will inherit the version of the docker cli rather than their own version.

From a more generic standpoint, I don't really trust the various `service.x` attributes very much because a user can override them by setting an environment variable. This is why I've preferred to use the instrumentation library version.

I wouldn't oppose us adding this though. You can use `WithResourceOptions` when creating the CLI to inject the service version in a similar way.

Reply:
It would still be useful for us to know what version of the CLI a plugin was compiled with though. Maybe using a different, non-env-overrideable attribute, sure.

Reply:
This makes sense to me. Plugins seem like a separate concern - if we get to `docker/cli` bootstrapping OTel for the plugin, we should probably have it create a resource _for_ the plugin as part of that, at which point it could use the plugin version from `Metadata` etc

Fwiw from the SDK side, I'd assume the env var would take lower precedence than anything explicitly defined in code? That's how e.g. `SERVICE_NAME` works IIRC.
(From a code perspective, when you call `NewResource()`, I think the env detector will run and THEN your overrides.)

Reply:
It looks like version name isn't overridable through an environment variable. It seems to be only [SERVICE_NAME](https://opentelemetry.io/docs/specs/semconv/resource/#semantic-attributes-with-dedicated-environment-variable).

I think it's ok if we include the version with `service.version`, but I'd like to make sure that the plugins don't inherit that code.

If we want plugins to know what version of the CLI they were invoked with, we can pass this through the environment variable that's being used for that: https://github.com/docker/cli/blob/85dcacd78f2a45886b1ab86ef49fe17014852add/cli-plugins/manager/manager.go#L245

We'd just have to modify that code to also pass the attributes from the resource.

I generally think that using the version that's included from [here](https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithInstrumentationVersion) should be enough. I don't think we need to set `service.version` at the moment.

Also, to my knowledge, the environment variable overrides anything defined in code.

Reply:
@milas also just thinking out loud, we could just not honor the `OTEL_SERVICE_NAME` environment variable. I think that environment variable is likely more helpful for actual services and not the CLI.

Reply:
Reverted the change for now, but I'll create a ticket so we can look at this later.